### ISSUE TITLE
Give more visibility to the documentation

### ISSUE THREAD

**Issue by OlivierBlanvillain** [typelevel/cats#1125]
I think the cats documentation is a bit hard to reach from the README, this is the expected path to [`SemigroupK`](https://github.com/typelevel/cats/blob/master/docs/src/main/tut/semigroupk.md):

- Land on https://github.com/typelevel/cats
- → click Documentation#website link
- → click Typeclasses
- → scroll down
- → click SemigroupK

As an example, I find it way more accessible on [`scalajs-react`](https://github.com/japgolly/scalajs-react) or  [`scalacss`](https://github.com/japgolly/scalacss) :

- Land on https://github.com/japgolly/scalajs-react
- → click Router

My suggestion would be to give it more visibility on the README, for instance by adding a big pile of links like the following:

[Documentation is here.](http://typelevel.org/cats/)

- [Typeclasses:](http://typelevel.org/cats/typeclasses.html)

    - [Applicative](http://typelevel.org/cats/tut/applicative.html)
    - [Apply](http://typelevel.org/cats/tut/apply.html)
    - [Contravariant](http://typelevel.org/cats/tut/contravariant.html)
    - [Faq](http://typelevel.org/cats/tut/faq.html)
    - [Foldable](http://typelevel.org/cats/tut/foldable.html)
    - [Functor](http://typelevel.org/cats/tut/functor.html)
    - [Id](http://typelevel.org/cats/tut/id.html)
    - [Invariant](http://typelevel.org/cats/tut/invariant.html)
    - [Monad](http://typelevel.org/cats/tut/monad.html)
    - [Monadcombine](http://typelevel.org/cats/tut/monadcombine.html)
    - [Monadfilter](http://typelevel.org/cats/tut/monadfilter.html)
    - [Monoid](http://typelevel.org/cats/tut/monoid.html)
    - [Monoidk](http://typelevel.org/cats/tut/monoidk.html)
    - [Semigroup](http://typelevel.org/cats/tut/semigroup.html)
    - [Semigroupk](http://typelevel.org/cats/tut/semigroupk.html)
    - [Show](http://typelevel.org/cats/tut/show.html)
    - [Traverse](http://typelevel.org/cats/tut/traverse.html)

- [Data Types:](http://typelevel.org/cats/datatypes.html)

    - [Const](http://typelevel.org/cats/tut/const.html)
    - [Freeapplicative](http://typelevel.org/cats/tut/freeapplicative.html)
    - [Freemonad](http://typelevel.org/cats/tut/freemonad.html)
    - [Kleisli](http://typelevel.org/cats/tut/kleisli.html)
    - [Oneand](http://typelevel.org/cats/tut/oneand.html)
    - [Optiont](http://typelevel.org/cats/tut/optiont.html)
    - [State](http://typelevel.org/cats/tut/state.html)
    - [Validated](http://typelevel.org/cats/tut/validated.html)
    - [Xor](http://typelevel.org/cats/tut/xor.html)

**Comment by ceedubs** [comment]
@OlivierBlanvillain I totally agree that our docs aren't very discoverable. @non has a branch where he has done a bit of cleanup, but we still have a long way to go. Part of our strategy with his branch was to have less noise on the readme so that the things that _are_ there are more noticeable.

It may be nice to have a big list of links like that, but I think there may be even better options. I think that #95 would provide the same information for the type classes but in a more digestible way (and would show the relationships between them). I also like the idea of something like the [book of doobie](http://tpolecat.github.io/doobie-0.2.3/00-index.html) that has a clear start point and progression for newcomers. People who are already familiar with cats are going to know how to get to the information that they want, so I am interested in focusing on people who _aren't_ already familiar with cats and don't know which of the 20 items in the list they should click on to start with.

Sorry this is kind of a brain dump. In short, I definitely think we need to improve our docs, so thank you for speaking up about it.

**Comment by OlivierBlanvillain** [comment]
For a result similar to the book of doobie, I think that Gitbook could be a really good candidate. In a [few lines of shell](https://github.com/jto/validation/blob/v2.0/misc/build-book.sh#L4-L11) you can go from https://github.com/jto/validation/tree/v2.0/docs/src/main/tut to http://jto.github.io/validation/docs/book/, it also has a search thing which works pretty well. If you think that's a good idea I could give it a try on cats.

A type classes diagram looks sound really great :smile:

**Comment by ceedubs** [comment]
@OlivierBlanvillain oh wow that looks pretty nice. I remember @adelbertc mentioning Gitbook at some point, but I never took the time to look into it. It looks like in your example your code in your book is still compiled via tut - is that right? This looks great to me!

**Comment by OlivierBlanvillain** [comment]
Yes, it's .tut → .md → .html :). The last step is done via an [open source node.js tool](https://github.com/GitbookIO/gitbook), https://www.gitbook.com/ makes it sound like it's all hosted but you can actually do everything locally.

**Comment by ceedubs** [comment]
@OlivierBlanvillain I realized that I never explicitly said this, but if you are interested in giving a cats git-book a try, I think that would be fantastic.

**Comment by yilinwei** [comment]
I really like the book of doobie, thanks for linking that!

I've been wanting to clean up the docs as well not just from a beginner perspective but as,

- migration/differences from scalaz
- library authors
- new contributors
- new users of FP

**Comment by zainab-ali** [comment]
Could we add experienced users of FP to that list? i.e. users who understand FP, but are new to the way it is implemented in cats.

**Comment by Jorgeptz** [comment]
> Creo que la documentación de gatos es un poco difícil de alcanzar desde el archivo README, este es el camino esperado para [`SemigroupK`](https://github.com/typelevel/cats/blob/master/docs/src/main/tut/semigroupk.md):
> 
> * Aterriza en https://github.com/typelevel/cats
> * → haga clic en el enlace del sitio web Documentation #
> * → haga clic en Typeclasses
> * → desplazarse hacia abajo
> * → haga clic en SemigroupK
> 
> Como ejemplo, me parece mucho más accesible en [`scalajs-react`](https://github.com/japgolly/scalajs-react)o [`scalacss`](https://github.com/japgolly/scalacss):
> 
> * Aterrice en https://github.com/japgolly/scalajs-react
> * → haga clic en Router
> 
> Mi sugerencia sería darle más visibilidad en el archivo README, por ejemplo, agregando una gran pila de enlaces como el siguiente:
> 
> [La documentación está aquí.](http://typelevel.org/cats/)
> 
> * [Clases de tipo:](http://typelevel.org/cats/typeclasses.html)
>   
>   * [Aplicativo](http://typelevel.org/cats/tut/applicative.html)
>   * [Aplicar](http://typelevel.org/cats/tut/apply.html)
>   * [Contravariante](http://typelevel.org/cats/tut/contravariant.html)
>   * [Preguntas más frecuentes](http://typelevel.org/cats/tut/faq.html)
>   * [Plegable](http://typelevel.org/cats/tut/foldable.html)
>   * [Functor](http://typelevel.org/cats/tut/functor.html)
>   * [Carné de identidad](http://typelevel.org/cats/tut/id.html)
>   * [Invariante](http://typelevel.org/cats/tut/invariant.html)
>   * [Monada](http://typelevel.org/cats/tut/monad.html)
>   * [Monadcombine](http://typelevel.org/cats/tut/monadcombine.html)
>   * [Monadfilter](http://typelevel.org/cats/tut/monadfilter.html)
>   * [Monoide](http://typelevel.org/cats/tut/monoid.html)
>   * [Monoidk](http://typelevel.org/cats/tut/monoidk.html)
>   * [Semigrupo](http://typelevel.org/cats/tut/semigroup.html)
>   * [Semigroupk](http://typelevel.org/cats/tut/semigroupk.html)
>   * [mostrar](http://typelevel.org/cats/tut/show.html)
>   * [atravesar](http://typelevel.org/cats/tut/traverse.html)
> * [Tipos de datos:](http://typelevel.org/cats/datatypes.html)
>   
>   * [Const](http://typelevel.org/cats/tut/const.html)
>   * [Aplicación gratuita](http://typelevel.org/cats/tut/freeapplicative.html)
>   * [Freemonad](http://typelevel.org/cats/tut/freemonad.html)
>   * [Kleisli](http://typelevel.org/cats/tut/kleisli.html)
>   * [Uno y](http://typelevel.org/cats/tut/oneand.html)
>   * [Optiont](http://typelevel.org/cats/tut/optiont.html)
>   * [Estado](http://typelevel.org/cats/tut/state.html)
>   * [Validado](http://typelevel.org/cats/tut/validated.html)
>   * [Xor](http://typelevel.org/cats/tut/xor.html)

**Comment by Jorgeptz** [comment]
Duplicado de #
